libtsuba/include/tsuba/RDG.h

```diff
@@ -63,41 +68,100 @@ class KATANA_EXPORT RDG {
   katana::Result<void> Store(
       RDGHandle handle, const std::string& command_line,
-      RDGVersioningPolicy versioning_action, std::unique_ptr<FileFrame> ff);
+      RDGVersioningPolicy versioning_action,
+      std::unique_ptr<FileFrame> topology_ff,
+      std::unique_ptr<FileFrame> node_entity_type_id_array_ff,
+      std::unique_ptr<FileFrame> edge_entity_type_id_array_ff,
+      const std::unordered_map<EntityTypeID, std::vector<EntityTypeID>>&
+          node_entity_type_id_dictionary,
+      const std::unordered_map<EntityTypeID, std::vector<EntityTypeID>>&
+          edge_entity_type_id_dictionary,
+      const std::unordered_map<EntityTypeID, std::vector<std::string>>&
+          node_entity_type_id_name,
+      const std::unordered_map<EntityTypeID, std::vector<std::string>>&
+          edge_entity_type_id_name);
 
   /// @brief Store new version of the RDG with lineage based on command line.
   /// @param handle :: handle indicating where to store RDG
   /// @param command_line :: added to metadata to track lineage of RDG
-  /// @param ff :: FileFrame for topology information, can be nullptr. If set, topology
-  /// information will be persisted to ff.
+  /// @param topology_ff :: if not nullptr, it is persisted as the topology for this RDG.
+  /// @param node_entity_type_id_array_ff :: if not nullptr, it is persisted as the node_entity_type_id_array for this RDG.
+  /// @param edge_entity_type_id_array_ff :: if not nullptr, it is persisted as the edge_entity_type_id_array for this RDG.
+  /// @param node_entity_type_id_dictionary :: persisted as the node EntityTypeID -> Atomic node EntityType id mapping
+  /// @param edge_entity_type_id_dictionary :: persisted as the edge EntityTypeID -> Atomic edge EntityType id mapping
+  /// @param node_entity_type_id_name :: persisted as the node EntityTypeID -> node EntityTypeID name mapping
+  /// @param edge_entity_type_id_name :: persisted as the edge EntityTypeID -> edge EntityTypeID name mapping
   katana::Result<void> Store(
       RDGHandle handle, const std::string& command_line,
-      std::unique_ptr<FileFrame> ff) {
-    return Store(handle, command_line, IncrementVersion, std::move(ff));
+      std::unique_ptr<FileFrame> topology_ff,
+      std::unique_ptr<FileFrame> node_entity_type_id_array_ff,
+      std::unique_ptr<FileFrame> edge_entity_type_id_array_ff,
+      const std::unordered_map<EntityTypeID, std::vector<EntityTypeID>>&
+          node_entity_type_id_dictionary,
+      const std::unordered_map<EntityTypeID, std::vector<EntityTypeID>>&
+          edge_entity_type_id_dictionary,
+      const std::unordered_map<EntityTypeID, std::vector<std::string>>&
+          node_entity_type_id_name,
+      const std::unordered_map<EntityTypeID, std::vector<std::string>>&
+          edge_entity_type_id_name) {
+    return Store(
+        handle, command_line, IncrementVersion, std::move(topology_ff),
```

Comment:
Should we really be incrementing version upon storing these arrays? @tylershunt . My understanding was that version should increment upon changes to the graph.

Reply:
the different use cases of the different overloaded `Store` functions isn't very clear to me, clarification of which ones should/shouldn't include these 6 structures would be be welcome. 

its not clear to me why overloading is preferable here in general vs writing wrapper functions with better defined names/tasks. That said, I'm also new to this codebase so take my opinion with a grain of salt :)

Reply:
Yeah the overloading here is unfortunate, not sure why we went with that. What's going on here (i think) is that we have two cases when we would write out an RDG:
  1. it's a new logical version of the graph so the version number needs to be bumped
  2. it's a a different view of the same logical graph so the version number should remain the same (and view info changes).
  
Would prefer just to have a `Store` and `StoreView` function rather than this cryptic enum. But I digress.

I think our changes here should be agnostic to this, in answer to @amberhassaan's question we increment the version upon write; in-memory updates don't get version numbers.